.github/workflows/test.yml

```diff
@@ -55,10 +55,75 @@ jobs:
             *.cache-to=type=gha,scope=dev${{ matrix.mode }},mode=max
             *.output=type=cacheonly
 
+  validate:
+    runs-on: ubuntu-20.04
+    needs:
+      - build-dev
+    strategy:
+      fail-fast: true
+      matrix:
+        script:
+          - dco
+          - default-seccomp
+          - pkg-imports
+          - yamllint
+          - swagger
+          - swagger-gen
+          - toml
+          - changelog-well-formed
+          - changelog-date-descending
+          - golangci-lint
+          - shfmt
+          - vendor
```

Comment:
I like the fact these will show up as separate checks in CI.

I'm wondering if there's some way we can think of to make sure we don't accidentally _miss_ validation checks if they would be added 🤔

I guess the same already applies currently, as we need to keep the `hack/validate/default` script updated, so (IMO) it's fine to look at that separately;  https://github.com/moby/moby/blob/924edb948c2731df3b77697a8fcc85da3f6eef57/hack/validate/default#L7-L19

Reply:
Speaking of which; it looks like we disabled `deprecate-integration-cli` in https://github.com/moby/moby/commit/7ed823ead91938e5cc8de7a42ff93a25abe73e7e, but forgot to re-enable it.

Reply:
Opened https://github.com/moby/moby/pull/44129 to re-enable that validation (probably should be updated in this list as well)

Reply:
> I'm wondering if there's some way we can think of to make sure we don't accidentally miss validation checks if they would be added 🤔

We could create a dynamic matrix sure.

Reply:
Done, see second commit (can squashed later)